src/core/main.c

```diff
@@ -3314,6 +3314,16 @@ int main(int argc, char *argv[]) {
 
         log_execution_mode(&first_boot);
 
+        r = cg_has_legacy();
+        if (r < 0) {
+                error_message = "Failed to check cgroup hierarchy";
+                goto finish;
+        }
+        if (r > 0) {
+                error_message = "Detected unsupported legacy cgroup hierarchy, refusing execution";
+                goto finish;
+        }
```

Comment:
If we cannot boot, I think we should at least provide easy advice how to modify the kernel command line to boot successfully.